src/components/Api.js

```diff
@@ -0,0 +1,212 @@
+        }
+      })
+      .then(values => {
+        const titleValue = values.name;
+        const urlValue = values.link;
+
+        const cardElement = createNewCard({name: titleValue, link: urlValue, _id: values._id});
+
+        document.querySelector('.elements').prepend(cardElement);
+        formValidatorCardModal.disableStateButton();
+      })
+      .catch(err => console.error(err))
+      .finally(() => {
+        this.renderLoading(false, popup);
+      })
+    }
+
+    deleteCard(cardId){
+      return fetch(`${this._baseUrl}/cards/${cardId}`, {
+        method: 'DELETE',
+        headers: this._headers
+      })
+      .then(res => {
+        if(res.ok){
+          return res.json();
+        } else{
+          return Promise.reject(`Error: ${res.status}`);
+        }
+      })
+      .catch(err => {
+        console.error(err);
+      })
+    }
+
+    addLike(cardId, cardElement){
+      return fetch(`${this._baseUrl}/cards/${cardId}/likes`, {
+        method: 'PUT',
+        headers: this._headers1
+      })
+      .then(res => {
+        if(res.ok){
+          return res.json();
+        } else{
+          return Promise.reject(`Error: ${res.status}`);
+        }
+      })
+      .then(res => {
+        if(res.isLiked){
+          cardElement.querySelector('.card__love-icon').classList.add('card__love-icon_background_black');
+        }
+      })
+      .catch(err => {
+        console.error(err);
+      })
+    }
+
+    deleteLike(cardId, cardElement){
+      return fetch(`${this._baseUrl}/cards/${cardId}/likes`, {
+        method: 'DELETE',
+        headers: this._headers1
+      })
+      .then(res => {
+        if(res.ok){
+          return res.json();
+        } else{
+          return Promise.reject(`Error: ${res.status}`);
+        }
+      })
+      .then(res => {
+        if(!res.isLiked){
+          cardElement.querySelector('.card__love-icon').classList.remove('card__love-icon_background_black');
+        }
+      })
+      .catch(err => {
+        console.error(err);
+      })
+    }
+
+    updateProfilePicture(inputValues, popup){
+      return fetch(`${this._baseUrl}/users/me/avatar`, {
+        method: 'PATCH',
+        headers: this._headers,
+        body: JSON.stringify(inputValues)
+      })
+      .then(res => {
+        if(res.ok){
+          return res.json();
+        } else{
+          return Promise.reject(`Error: ${res.status}`);
+        }
+      })
+      .then(values => {
+        document.querySelector('.profile__image').src = values.avatar;
+        console.log(values);
+      })
+      .catch(err => console.error(err))
+      .finally(() => {
+        this.renderLoading(false, popup);
+      })
+    }
+
+    renderLoading(isLoading, popup){
+        if(isLoading){
+            popup.querySelector('.modal__button').textContent = 'Saving...';
+        } else{
+            popup.querySelector('.modal__button').textContent = 'Save';
+        }   
```

Comment:
`renderLoading` should not be inside `Api` because it's only for handling api (not rendering loading texts)